bpf/lib/lb.h

```diff
@@ -1122,12 +1157,41 @@ static __always_inline int lb6_local(const void *map, struct __ctx_buff *ctx,
 		lb6_update_affinity_by_addr(svc, &client_id, backend_id);
 #endif
 
-#if defined(ENABLE_LOCAL_REDIRECT_POLICY)
-	if (netns_cookie > 0 && unlikely(lb6_svc_is_localredirect(svc)) &&
-	    lb6_skip_xlate_from_ctx_to_svc(netns_cookie, tuple->daddr, tuple->sport))
-		return CTX_ACT_OK;
+#if defined(ENABLE_LOCAL_REDIRECT_POLICY) || defined (USE_LOOPBACK_LB)
+	if (saddr.addr[0] == backend->address.addr[0] &&
+	    saddr.addr[1] == backend->address.addr[1] &&
+	    saddr.addr[2] == backend->address.addr[2] &&
+	    saddr.addr[3] == backend->address.addr[3] &&
+	    saddr.addr[4] == backend->address.addr[4] &&
+	    saddr.addr[5] == backend->address.addr[5] &&
+	    saddr.addr[6] == backend->address.addr[6] &&
+	    saddr.addr[7] == backend->address.addr[7] &&
+	    saddr.addr[8] == backend->address.addr[8] &&
+	    saddr.addr[9] == backend->address.addr[9] &&
+	    saddr.addr[10] == backend->address.addr[10] &&
+	    saddr.addr[11] == backend->address.addr[11] &&
+	    saddr.addr[12] == backend->address.addr[12] &&
+	    saddr.addr[13] == backend->address.addr[13] &&
+	    saddr.addr[14] == backend->address.addr[14] &&
+	    saddr.addr[15] == backend->address.addr[15]) {
+		#if defined(ENABLE_LOCAL_REDIRECT_POLICY)
+			if (netns_cookie > 0 && lb6_svc_is_localredirect(svc) &&
+		    	lb6_skip_xlate_from_ctx_to_svc(netns_cookie, tuple->saddr, tuple->sport))
+				return CTX_ACT_OK;
+		#endif
+
+		#if defined(USE_LOOPBACK_LB)
+			union v6addr loopback_addr = IPV6_LOOPBACK;
+			ipv6_addr_copy(&new_saddr, &loopback_addr);
+			state->loopback = 1;
+		#endif
```

Comment:
I don't think this will pass the compiler. Note that you can extend `bpf/tests/tc_nodeport_test.c` as needed to iterate faster.